src/Diagnostics/Diagnostics.jl

```diff
@@ -38,34 +38,38 @@ using ..Writers
 using CLIMAParameters
 using CLIMAParameters.Planet: planet_radius
 struct EarthParameterSet <: AbstractEarthParameterSet end
-const param_set = EarthParameterSet()
 
 Base.@kwdef mutable struct Diagnostic_Settings
     mpicomm::MPI.Comm = MPI.COMM_WORLD
     dg::Union{Nothing, DGModel} = nothing
     Q::Union{Nothing, MPIStateArray} = nothing
     starttime::Union{Nothing, String} = nothing
+    param_set::Union{Nothing, AbstractEarthParameterSet} = nothing
     output_dir::Union{Nothing, String} = nothing
 end
 const Settings = Diagnostic_Settings()
 
 """
-    init(mpicomm, dg, Q, starttime, output_dir)
+    init(mpicomm, dg, Q, starttime, output_dir, param_set)
 
 Initialize the diagnostics collection module -- save the parameters into
 `Settings`.
+
+TODO: make `param_set` required!
 """
 function init(
     mpicomm::MPI.Comm,
     dg::DGModel,
     Q::MPIStateArray,
     starttime::String,
     output_dir::String,
+    param_set::AbstractEarthParameterSet = EarthParameterSet(),
```

Comment:
This should not be optional (it should _always_ be defined in the driver).

Reply:
Yes, it has to be passed in. But I realized that's not happening and set this up as a first step.